streams/src/main/java/org/apache/kafka/streams/kstream/internals/ChangedDeserializer.java

```diff
@@ -35,8 +36,11 @@ public ChangedDeserializer(final Deserializer<T> inner) {
         return inner;
     }
 
-    public void setInner(final Deserializer<T> inner) {
-        this.inner = inner;
+    @Override
+    public void setIfUnset(final Deserializer<T> defaultDeserializer) {
+        if (inner == null) {
+            inner = Objects.requireNonNull(defaultDeserializer, "defaultDeserializer cannot be null");
+        }
```

Comment:
Having generalized this concept, I'm switching Changed(De)Serializer over to the general strategy.